test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -213,6 +213,23 @@ object Assertion {
   final def containsString(element: String): Assertion[String] =
     Assertion.assertion("containsString")(param(element))(_.contains(element))
 
+  final def containsTheSameElements[A](other: Iterable[A]): Assertion[Iterable[A]] = {
```

Comment:
Could we just do `actual.diff(other).isEmpty && other.diff(actual).isEmpty`? That might fit well with how we want to report the cause of the assertion failure too.